config/config.go

```diff
@@ -55,6 +61,27 @@ func (d DatadogConfig) IsDataDogConfigValid() bool {
 	return true
 }
 
+// Validate the DatadogConfig. Returns the first error found, returns nil if
+// the configuration is good.
+func (d DatadogConfig) Validate() error {
+	if d.Env == "" {
+		return errors.New("DD_ENV must be defined")
+	}
+	if d.Service == "" {
+		return errors.New("DD_SERVICE must be defined")
+	}
+	if d.ServiceVersion == "" {
+		return errors.New("DD_VERSION must be defined")
+	}
+
+	// DSD or APM must be configured`
+	if d.DSD == "" && d.APM == "" {
+		return errors.New("DD_DOGSTATSD_URL and/or DD_TRACE_AGENT_URL must be defined")
+	}
+
+	return nil
+}
```

Comment:
@bendiknesbo I'm not sure if we should collect all the errors or return them one by one like this.

Reply:
It's a lower cognitive load to return them one by one, and if there is an error in one, the developers should review their full config anyway, so I say: keep it as is.